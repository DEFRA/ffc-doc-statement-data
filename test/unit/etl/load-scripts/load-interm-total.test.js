const db = require('../../../../app/data')
const { loadIntermTotal } = require('../../../../app/etl/load-scripts/load-interm-total')

jest.mock('../../../../app/data')

describe('loadIntermTotal', () => {
  let mockTransaction

  beforeEach(() => {
    mockTransaction = {} // Mock transaction object
    db.sequelize.query.mockClear()
  })

  test('should call sequelize.query with correct parameters', async () => {
    const startDate = new Date('2023-01-01')

    await loadIntermTotal(startDate, mockTransaction)

    const expectedQuery = `
    INSERT INTO etl_interm_total (
      payment_ref, quarter, total_amount,
      transdate
    )
    SELECT payment_ref, 
      D.quarter,
      SUM(transaction_amount) * -1 as total_amount,
      transdate 
    FROM etl_interm_finance_dax D 
    WHERE D.payment_ref LIKE 'PY%' 
      AND D.etl_inserted_dt > :startDate
    GROUP BY transdate, quarter, payment_ref
    ORDER BY payment_ref;
  `

    expect(db.sequelize.query).toHaveBeenCalledWith(
      expectedQuery,
      {
        replacements: { startDate },
        raw: true,
        transaction: mockTransaction
      }
    )
  })
})
