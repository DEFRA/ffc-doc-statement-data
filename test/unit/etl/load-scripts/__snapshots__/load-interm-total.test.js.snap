// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadIntermTotal should call sequelize.query with correct parameters 1`] = `
[MockFunction] {
  "calls": [
    [
      "
    INSERT INTO etlIntermTotal (
      paymentRef, quarter, totalAmount,
      transdate, invoiceid
    )
    SELECT DISTINCT paymentRef, 
      D.quarter,
      SUM(transactionAmount) * -1 as totalAmount,
      transdate,
      invoiceid
    FROM "etlIntermFinanceDax" D 
    WHERE D.paymentRef LIKE 'PY%' 
      AND D.etlInsertedDt > :startDate
    GROUP BY transdate, quarter, paymentRef, invoiceid
    ORDER BY paymentRef;
  ",
      {
        "raw": true,
        "replacements": {
          "startDate": 2023-01-01T00:00:00.000Z,
        },
        "transaction": {},
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
